src/dmd/mtype.d

```diff
@@ -7108,6 +7108,11 @@ extern (C++) final class Parameter : ASTNode
 
     extern (D) private static bool isCovariantScope(bool returnByRef, StorageClass from, StorageClass to) pure nothrow @nogc @safe
     {
+        if ((to & STC.scopeinferred) && !(to & STC.return_))
```

Comment:
This is because `void f(return int* x)` gets turned into `void f(return $scope int* x)`, and there we don't want to ignore `scope`.